modules/fx/arrow-fx/src/main/kotlin/arrow/fx/extensions/QueueInvariant.kt

```diff
@@ -0,0 +1,29 @@
+package arrow.fx.extensions
+
+import arrow.Kind
+import arrow.extension
+import arrow.fx.Queue
+import arrow.fx.QueueOf
+import arrow.fx.QueuePartialOf
+import arrow.fx.fix
+import arrow.typeclasses.Functor
+import arrow.typeclasses.Invariant
+import arrow.undocumented
+
+@extension
+@undocumented
+interface QueueInvariant<F> : Invariant<QueuePartialOf<F>> {
+  fun FR(): Functor<F>
+
+  override fun <A, B> QueueOf<F, A>.imap(f: (A) -> B, g: (B) -> A): Queue<F, B> =
+    FR().run {
```

Comment:
Android had problems with this pattern and you had to put the `run` block as close as the usage as possible. Hopefully it's okay now.